pkg/scheduler/scheduler.go

```
@@ -240,13 +242,19 @@ func (s *Scheduler) assign(ctx context.Context, e *entry) error {
 // given that wUsed is the usage of flavors by previous podsets.
 // If it finds a flavor, also returns any borrowing required.
 func findFlavorForResource(name corev1.ResourceName, val int64, wUsed map[string]int64, cap *capacity.Capacity, spec *corev1.PodSpec) (string, int64) {
+	// We will only check against the flavors' labels for the resource.
+	selector := flavorSelector(spec, cap.LabelKeys[name])
 	for _, flavor := range cap.RequestableResources[name] {
 		_, untolerated := corev1helpers.FindMatchingUntoleratedTaint(flavor.Taints, spec.Tolerations, func(t *corev1.Taint) bool {
 			return t.Effect == corev1.TaintEffectNoSchedule || t.Effect == corev1.TaintEffectNoExecute
 		})
 		if untolerated {
 			continue
 		}
+		if match, _ := selector.Match(&corev1.Node{ObjectMeta: metav1.ObjectMeta{Labels: flavor.Labels}}); !match {
```

Fail instead of ignoring the error. Scheduler would also fail.